h/models/token.py

```diff
@@ -48,10 +58,13 @@ class Token(Base, mixins.Timestamps):
     #: A NULL value means it is a developer token.
     authclient = sqlalchemy.orm.relationship('AuthClient')
 
-    def __init__(self, **kwargs):
-        super(Token, self).__init__(**kwargs)
+    def __init__(self, expires=None, **kwargs):
```

Comment:
I think that SQLAlchemy models have a specific constructor which we shouldn't change. Adding the optional `expires` is okay, but it doesn't add anything because you can just read it from `kwargs`. I wouldn't make the `userid` a required parameter though, because it adds complexity that needs to be kept in our heads of how the constructors look like for the different models. And since we have a database-level constraint that checks that it is not missing there isn't a data integrity issue. The error message is also pretty clear stating that the value of the required column is missing.